presto-parser/src/main/java/com/facebook/presto/sql/TemplateExpressionFormatter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql;
+
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.TemplateExpressionRewriter;
+
+import java.util.List;
+import java.util.Optional;
+
+public class TemplateExpressionFormatter
+        extends ExpressionFormatter
+{
+    @Override
+    public String format(Expression expression, Optional<List<Expression>> parameters)
+    {
+        return new Formatter(Optional.empty()).process(TemplateExpressionRewriter.rewrite(expression), null);
```

Comment:
It's a bit dangerous to just drop "parameters" and pass in Optional.empty(). Should we throw an exception if this method is called with parameters being not empty so the caller is aware they are doing something wrong? Or maybe we should handle this gracefully and just replace the non-parameter literals? Either way, let's make sure we add a test for this behavior, so the test can act as documentation for what this class is intended to do.

Reply:
Updated to throw an exception if this method is called with parameters not being empty.